CONTRIBUTING.md

```diff
@@ -0,0 +1,193 @@
+
+
+### 3. Comments
+Comments can be a blessing or a cure. Proper documentation can help developers while bad documentation can obstruct or mislead them.
+
+#### 3.1 Useless comments
+Ideal code is self explanatory. Do not comment when it is obvious like
+
+```
+//Initialize ROS
+initRos();
+```
+
+This adds lines for the programmers to read without adding any value.
+
+#### 3.2 Rotted Comments
+One major issue with comments is that they rot with time. Before writing a comment, make sure that this comment will still be relevant after the code passes through a lot of changes.
+
+#### 3.3 Todos
+Do not write any `Todo` comments. These get easily lost and forgotten. Open an issue instead.
+
+#### 3.4 Commented code
+Avoid commenting code. A lot of the code that is commented, never gets uncommented again. It is a better practice to remove the code. We are using git as a version control tool, this means that old code can always be recuperated.
+ 
+
+### 4. Meaningful Names
+
+#### 4.1 Variable Names
+
+With the function that we improved above there is still a glaring issue :
+
+```
+void Commands::setTto(bool throttleTimeOut) {
+    String msg = "ASTRO Throttle Timeout " + String(throttleTimeOut ? "On" : "Off");
+    println(msg);
+
+}
+```
+
+What the heck is `TTO`? By observing this function and it's argument, it is obvious that it's for `Throttle Timeout`. However, imagine that you were stumbling upon a call to this function like this
+
+```
+//some code
+setTto(false);
+//more code
+```
+To know what `TTO` means, you need to go to the function definition which could be tedious if you have to do this often.
+
+Furthermore, you would expect `throttleTimeOut` to be a number, the timeout time, but it is a boolean.
+
+```
+void Commands::hasThrottleTimeout(bool hasThrottleTimeout) {
+    String msg = "ASTRO Throttle Timeout " + String(hasThrottleTimeout ? "On" : "Off");
+    println(msg);
+}
+```
+
+Boolean variable should start with `has` or `is`.
+
+
+```
+//some code
+hasThrottleTimeout(false);
+//more code
+```
+Now it is much cleaner when the call is being read.
+
+#### 4.2 Magic Values
+
+Magic values (or magic numbers) are values used with unexplained context.
+
+An example of the usage of a magic value :
+```
+if (isPressed(input[65]))
+```
+
+We do not know what `65` is.
+
+To fix this problem, we can simply defined a constant.
+
+```
+if (isPressed(input[A_KEY]))
+```
+
+
+Magic values should be avoided and replaced by variables or constants with names that self-describes their behavior.
+
+
+
+## Programming Specific Guidelines
+
+### Python
+
+We use the PEP8 Style Guide. Please refer to it [here](https://www.python.org/dev/peps/pep-0008/)
+
+### JavaScript
+
+We are attempting to use a strict variant of JavaScript. We have not enable `strict mode` yet, but all future code should respect the strict variant of JavaScript which can be found [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode)
```

Comment:
`strict mode` was introduced in ECMAScript5 but we try to adhere to ECMAScript6 as per the first line [here](https://github.com/space-concordia-robotics/robotics-prototype#contributing-and-development-environment-instructions)

Reply:
I will also discuss that we use Standard JS from here as explained on the Wiki https://github.com/standard/standard/blob/master/RULES.md

Reply:
done

Reply:
yeah this is what we try to enforce with the `standard-formatter` in atom